packages/cli-e2e/__tests__/vue.specs.ts

```diff
@@ -0,0 +1,113 @@
+import {spawn, spawnSync} from 'child_process';
+
+import type {ChildProcessWithoutNullStreams} from 'child_process';
+
+import {CLI_EXEC_PATH, killCliProcessFamily} from '../utils/cli';
+import {closeAllPages, getBrowser} from '../utils/browser';
+import {Browser} from 'puppeteer-core';
+import {loginWithOffice} from '../utils/login';
+
+describe('ui', () => {
+  describe('create:vue', () => {
+    let browser: Browser;
+    const cliProcesses: ChildProcessWithoutNullStreams[] = [];
+    // TODO: CDX-90: Assign a dynamic port for the search token server on all ui projects
+    const clientPort = '8080';
+    const projectName = 'vue-project';
+    const searchPageEndpoint = `http://localhost:${clientPort}`;
+    const tokenProxyEndpoint = `http://localhost:${clientPort}/token`;
+    let waitForProjectToStartTimeout: NodeJS.Timeout;
+
+    beforeAll(async () => {
+      browser = await getBrowser();
+
+      await loginWithOffice(browser, cliProcesses);
+
+      const buildProcess = spawnSync(CLI_EXEC_PATH, [
+        'ui:create:vue',
+        projectName,
+      ]);
+
+      expect(buildProcess.status).toEqual(0);
+
+      const waitForProjectToStart = new Promise<void>((resolve) => {
+        const startServerProcess = spawn('npm', ['run', 'start'], {
+          cwd: projectName,
+          detached: true,
+        });
+
+        cliProcesses.push(startServerProcess);
+
+        waitForProjectToStartTimeout = setTimeout(() => {
+          resolve();
+        }, 15e3);
+      });
+
+      return waitForProjectToStart;
+    }, 240e3);
+
+    afterAll(async () => {
+      clearTimeout(waitForProjectToStartTimeout);
+      await browser.close();
+      return Promise.all(
+        cliProcesses.map((cliProcess) => killCliProcessFamily(cliProcess))
+      );
+    }, 5e3);
+
+    afterEach(async () => {
+      const pageClosePromises = await closeAllPages(browser);
+      return Promise.all(pageClosePromises);
+    }, 5e3);
+
+    it('should contain a search page section', async () => {
+      const page = await browser.newPage();
+      await page.goto(searchPageEndpoint, {
+        waitUntil: 'networkidle2',
+      });
+
+      expect(await page.$('#search-page')).not.toBeNull();
+    });
+
+    it('should retrieve the search token on the page load', async () => {
+      const page = await browser.newPage();
+      page.goto(searchPageEndpoint);
+      const tokenResponse = await page.waitForResponse(tokenProxyEndpoint);
+      expect(JSON.parse(await tokenResponse.text())).toMatchObject({
+        token: expect.stringMatching(/.+/),
+      });
+    });
+
+    it('should trigger search queries', async () => {
+      const page = await browser.newPage();
+      const searchboxSelector = '#search-page .autocomplete input';
+      let isInterfaceLoadSearch = true;
+
+      page.on('request', (request) => {
```

Comment:
I would do something a bit more crude maybe for the request interception but that would help with the reading of the test:
- Push all the requests in an array on the hook.

Then, using an async-retry do something like
`requests.some(isSearchRequest)` with `iSearchRequest being a function taking a request in param and returning true or false.

I would do it _once_ before interacting with the page, to check for the initial query.
Then, I would clear the array and do the search.
And do the requests.some again.

Why I'd do it like this?

Well, I think that setting up the hook on request is a 'setup' phase and should be done early and cleanly without any filtering. This makes clear that 'OK we listen on request for the test'.

Then you do an action:
`await page.goto(searchPageEndpoint, {waitUntil: 'networkidle2'});`
and right after you assert that there should be a request made to Platform.

Like this, the action and what it causes are close together and clear to the reader/tester/developer.

Reply:
oh I did not think about it. That's way cleaner 👍 .
I'm just not sure why we would need the async retry here.
Since we have specified the `waitUntil:"networkidle2"` option, we should be able to read all the intercepted requests right after the `page.goto` line.

As it goes for the second reading,  we can simply add an
`await page.waitForNetwork({ waitUntil: "networkidle0" });` to make sure we wait after all the search requests.

Reply:
Oh nvm. we need the async retry for the second reading :P